postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -796,4 +788,361 @@ process_file_bteq(char *filename, bool use_relative_path)
 
     pset.inputfile = oldfilename;
     return result;
+}
+
+
+/*
+ * .c or .connect -- connect to database using the specified parameters.
+ *
+ * .c [-reuse-previous=BOOL] dbname user host port
+ *
+ * Specifying a parameter as '-' is equivalent to omitting it.  Examples:
+ *
+ * .c - - hst        Connect to current database on current port of
+ *                    host "hst" as current user.
+ * .c - usr - prt    Connect to current database on port "prt" of current host
+ *                    as user "usr".
+ * .c dbs            Connect to database "dbs" on current port of current host
+ *                    as current user.
+ */
+static dotResult
+exec_command_connect(BteqScanState scan_state, bool active_branch)
```

Comment:
Why is this functionality added in scope of current PR?
As far as I know, .SET WIDTH command is not related to the database connection.

Reply:
it was added to allow us to join database to use tables